src/Security/SecurityConfigUpdater.php

```diff
@@ -149,6 +144,27 @@ private function updateEncoders(UserClassConfiguration $userConfig, string $user
         $this->manipulator->setData($newData);
     }
 
+    private function removeMemoryProviderIfIsSingleConfigured()
+    {
+        if (!$this->isSingleInMemoryProviderConfigured()) {
+            return;
+        }
+
+        $newData = $this->manipulator->getData();
+
+        $memoryProviderName = array_keys($newData['security']['providers'])[0];
+
+        $newData['security']['providers'] = [];
+
+        foreach ($newData['security']['firewalls'] as &$firewall) {
+            if (($firewall['provider'] ?? null) === $memoryProviderName) {
+                $firewall['provider'] = 'app_user_provider';
```

Comment:
👍 For the record, we prefer setting the provider explicitly. See https://github.com/symfony/recipes/pull/741